php/datasource/class-fieldmanager-datasource-user.php

```diff
@@ -44,22 +57,56 @@ class Fieldmanager_Datasource_User extends Fieldmanager_Datasource {
 
     /**
      * @var string|Null
-     * If not empty, set this post's ID as a value on the user. This is used to
+     * If not empty, set this object's ID as a value on the user. This is used to
      * establish two-way relationships.
      */
     public $reciprocal = Null;
 
-    // constructor not required for this datasource; options are just set to keys,
-    // which Fieldmanager_Datasource does.
+    /**
+	 * Constructor. Used for validation.
+	 */
+	public function __construct( $options = array() ) {
+		parent::__construct( $options );
+		
+		// Validate improper usage of store property
+		if ( ! in_array( $this->store_property, $this->allowed_store_properties ) ) {
+			throw new FM_Developer_Exception( sprintf( 
+				__( 'Store property %s is invalid. Must be one of %s.', 'fieldmanager' ),
+				$this->store_property,
+				implode( ', ', $this->allowed_store_properties )
+			) );
+		}
+		
+		if ( ! empty( $this->reciprocal ) && 'ID' != $this->store_property ) {
+			throw new FM_Developer_Exception( __( 'You cannot use reciprocal relationships with FM_Datasource_User if store_property is not set to ID.', 'fieldmanager' ) );
+		}
+	}
 
     /**
-     * Get a post title by post ID
+     * Get a user by the specified field.
      * @param int $value post_id
-     * @return string post title
+     * @return WP_User
```

Comment:
Or an `int`, I guess. Maybe worth just callign it `mixed` Regardless, worth adding a description that it returns whatever `display_property` is